components/character_icon.lua

```diff
@@ -0,0 +1,82 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:CharacterIcon
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local DateExt = require('Module:Date/Ext')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+
+local Data = Lua.requireIfExists('Module:CharacterIcon/Data', {loadData = true})
+
+---@class IconArguments
+---@field character string
+---@field size string?
+---@field class string?
+---@field date string?
+
+---@class CharacterIconInfo
+---@field file string
+---@field link string?
+---@field display string?
+---@field startDate string?
+---@field endDate string?
+
+local CharacterIcon = {}
+
+---@param icons CharacterIconInfo[]
+---@param date string?
+---@return CharacterIconInfo
+function CharacterIcon._getCharacterIconInfo(icons, date)
+	date = date or DateExt.getContextualDateOrNow()
+	local timeStamp = DateExt.readTimestamp(date)
+	local info = Array.find(icons, function (icon)
+		local startDate = DateExt.readTimestamp(icon.startDate) or DateExt.minTimestamp
+		local endDate = DateExt.readTimestamp(icon.endDate) or DateExt.maxTimestamp
+		return timeStamp >= startDate and timeStamp < endDate
+	end)
+	return info
+end
+
+---@param info CharacterIconInfo
+---@param size string?
+---@param class string?
+---@return string
+function CharacterIcon._makeImage(info, size, class)
+	local imageOptions = {
```

Comment:
imo use Array.append

```lua
local imageOptions = Array.append({
info.file,
info.display,
},
Logic.isNotEmpty(info.link) and ('link=' .. info.link) or nil,
Logic.isNotEmpty(class) and ('class=' .. class) or nil,
)
```
please apologize the bad formatting, i am on phone

Reply:
Keep table insert is better, if we have nil in array like `{'oi', nil, 'hello'}` table.concat will rip.

Reply:
append takes care of nils getting removed

Reply:
It works :D